[Merged by Bors] - Added Getting started guide (#225)
stackabletech/hive-operator: docs/modules/getting_started/pages/first_steps.adoc

= First steps

After going through the xref:installation.adoc[] section and having installed all the operators, you will now deploy a Hive metastore cluster and it's dependencies. Afterwards you can <<_verify_that_it_works, verify that it works>>.

== Setup

Several requirements should have already been installed in the xref:installation.adoc[Installation guide]:

* xref:commons-operator::index.adoc[Commons Operator]
* xref:secret-operator::index.adoc[Secret Operator]
* xref:hive-operator::index.adoc[Hive Operator]
* PostgreSQL
* MinIO for S3

=== S3Connection

In order to connect Hive to MinIO we need to create several files (or concat in one file).

> A S3Connection to connect to MinIO (_hive-minio-s3-connection.yaml_)

Reviewer: ```suggestion
An xref:home:concepts:s3.adoc[S3Connection] to connect to MinIO (`hive-minio-s3-connection.yaml`)
```

[source,yaml]
----
include::example$code/hive-minio-s3-connection.yaml[]
----

> Credentials for the S3Connection to log into MinIO (_hive-minio-credentials.yaml_)

Reviewer: ```suggestion
Credentials for the S3Connection to log into MinIO (`hive-minio-credentials.yaml`)
```

[source,yaml]
----
include::example$code/hive-minio-credentials.yaml[]
----

> A SecretClass for the credentials (hive-minio-credentials-secret-class.yaml)

Reviewer: ```suggestion
A SecretClass for the credentials to the Minio. The credentials were defined in the installation of Minio (`hive-minio-credentials-secret-class.yaml`)
```

[source,yaml]
----
include::example$code/hive-minio-credentials-secret-class.yaml[]
----

> The Apache Hive cluster definition (_hive-postgres-s3.yaml_)

Reviewer: ```suggestion
And lastly the actual Apache Hive cluster definition (`hive-postgres-s3.yaml`). Note how it references the `minio` S3 object
```

[source,yaml]
----
include::example$code/hive-postgres-s3.yaml[]
----

Once the files are created apply them to the cluster:

[source,bash]
----
include::example$code/getting-started.sh[tag=install-hive]
----

== Verify that it works

Make sure that all the Pods in the StatefulSets are ready:

[source,bash]
----
kubectl get statefulset
----

The output should show all pods ready:

[source,bash]
----
NAME                                 READY   AGE
hive-postgres-s3-metastore-default   1/1     48s
----

> For further testing we recommend to use e.g. the python https://github.com/quintoandar/hive-metastore-client[hive-metastore-client] in order to e.g. create tables or connect a product like Trino via the xref:trino-operator::index.adoc[Stackable Trino Operator].

Reviewer: ```suggestion
For further testing we recommend to use e.g. the python https://github.com/quintoandar/hive-metastore-client[hive-metastore-client] in order to e.g. create tables or connect a product like Trino via the xref:trino::index.adoc[Stackable Trino Operator].
```
